pages/employee/reports.vue

```diff
@@ -135,6 +136,36 @@
         viewTime.value = viewTimeNum + " h";
         viewTimeEarnings.value = Math.floor(viewTimeEarningsNum) + ' PLN';
     })
+
+    const generate = async () => {
+        await nextTick();
+
+        const startDate = new Date(2023, selectedMonth.value.value - 1, 1);
+        const endDate = new Date(2023, selectedMonth.value.value, 0);
+        
+        const {data,pending,error,refresh} = await useFetch(baseAPIURL + "/monthlyReportCompany", {
+            method: 'POST', body: {
+                "creationDate": new Date().toISOString(),
+                "startDate": startDate.toISOString(),
+                "endDate": endDate.toISOString(),
+                "viewers": views.value,
```

Comment:
wybierając wrzesień generuje raport od 31.08 - to najpewniej wina `toISOString`

Tworząc datę wyżej ustawiana jest godzina 00:00 w naszej strefie czasowej a w czasie ISO to 23:00. trzeba dodać do new Date ustawianie godziny